processing/src/main/java/io/druid/query/lookup/LookupBean.java

```diff
@@ -19,32 +19,84 @@
 
 package io.druid.query.lookup;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+
+import java.util.Objects;
 
-//TODO merge this code to the same definition when pr/1576 is merged
 class LookupBean
 {
+  private final LookupExtractorFactoryContainer container;
+  private final String name;
+
+  //to support rollback from 0.10.0 to 0.9.2 if necessary
+  @Deprecated
+  private final LookupExtractorFactory factory;
+
+
+  @JsonCreator
+  public LookupBean(
+      @JsonProperty("name") String name,
+      //kept for backward compatibility with druid ver <= 0.9.2 persisted snapshots
```

Comment:
updated the comment but don't see the need to make it a javadoc, this is not intended to users but devs only looking at the code